_posts/2020-07-06-hhvm-4.65.markdown

```diff
@@ -0,0 +1,41 @@
+---
+title: "HHVM 4.65"
+layout: post
+author: fred
+category: blog
+---
+
+HHVM 4.65 is released! This release marks the end of support for 4.58;
+HHVM 4.59&ndash;4.64 remain supported, as do the 4.32
+and 4.56 LTS releases.
+
+# Highlights
+
+- `is` and `as` checks for reified enums now only check the underlying type,
+  not the values; this makes them consistent with parameter and return type
+  enforcement.
+- added `_with_error()` variants of preg functions, e.g.
+  `preg_grep_with_error()`, `preg_match_with_error()`,
+  `preg_replace_with_error()`, to avoid the need for `preg_last_error()` calls
+  and the global state that entails.
+- added `HH\is_vec_or_varray()` and `HH\is_dict_or_darray()` functions.
+- clearer error messages when incompatible kinds `extend` each other - for
+  example, a class can not `extend` an interface, but it can `implement` one.
+
+# Breaking Changes
+
+- `array()` literals are no longer supported by the parser; `[]` literals were
+  already banned. Replace with `darray[]` or `varray[]`.
+- the globals array has been removed - for example, `HH\global_get('GLOBALS')`
+  now returns null. Fetch specific known keys instead.
```

Comment:
Probably worth mentioning that `$GLOBALS['some_key']` still works (or doesn't it)?

Reply:
That seems like an oversight to me - will follow up:

```
hphpd> =$GLOBALS['foo']
=$GLOBALS['foo']
null
hphpd> =$GLOBALS
=$GLOBALS
Hit a php exception : Error: Access $GLOBALS via wrappers in :1
hphpd> =$waefawef['foo']
=$waefawef['foo']

Notice: Undefined variable: waefawef
null
```